### DISCUSSION TITLE
Layout of PTA plots

### DISCUSSION THREAD

**Discussion by livPKPD** [LAPKB/Pmetrics#95] [General]
Hi there, I had a nice layout of 6 different PTA plots and a legend, which i did using layout() in base R graphics.
I then re-did the PTA analysis a few day later and the plots are now done in plotly, which is much harder (for me) to figure out how to present a grid or layout of all the plots. I have tried using the subplot() function in plotly but I am finding this challenging to figure out. Can I revert to the PTA plots I was getting last week?!

Thanks

**Comment by mnneely** [comment]
You should be able to use plot.PMpta(x), and hand it the pta object yes. Check out my Article on plotly on the new website at https://lapkb.github.io/Pmetrics/.

It allows for a lot more control over the graph, but does take a bit of learning. If you find that it can’t do something you’d like, let me know and I’ll get it added.

From: livPKPD ***@***.***>
Date: Friday, September 30, 2022 at 4:51 PM
To: LAPKB/Pmetrics ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [LAPKB/Pmetrics] Layout of PTA plots (Discussion #95)

Hi there, I had a nice layout of 6 different PTA plots and a legend, which i did using layout() in base R graphics.
I then re-did the PTA analysis a few day later and the plots are now done in plotly, which is much harder (for me) to figure out how to present a grid or layout of all the plots. I have tried using the subplot() function in plotly but I am finding this challenging to figure out. Can I revert to the PTA plots I was getting last week?!

Thanks

**Reply by livPKPD** [reply]
Thanks - I can't figure out how to specify the colours and line type on a line plot (PTA) with 9 different lines. I would also like to plot out the 6 different plots in a grid with a single common legend. Alternatively, can I just go back to the old school way of plotting PTA? I had it all coded nicely but I guess I must have updated PMetrics since!

**Comment by mnneely** [comment]
Sorry, I wasn’t clear enough. My last post told you how to plot with the legacy method.

Use plot.PMpta(x) where x is the result of makePTA.  If you made it with R6 using $pta() method, e.g. pta1 <- sim$pta(…), then use plot.PMpta(pta1$data). The $data field in all Pmetrics R6 objects contains the legacy object, in this case the output of makePTA.

Hope that makes sense. There’s always going to be a bit of pain transitioning to a new framework, but I think eventually you will find it more stable and logical. However, we intend to maintain the old style for those who simply prefer it.

If you post a picture of what you want the graph to look like, I can probably tell you how to do it in R6/plotly as a learning experience.
________________________________
